graphbolt/src/csc_sampling_graph.cc

```diff
@@ -385,9 +420,17 @@ int64_t NumPick(
     int64_t num_neighbors) {
   int64_t num_valid_neighbors =
       probs_or_mask.has_value()
-          ? *torch::count_nonzero(
-                 probs_or_mask.value().slice(0, offset, offset + num_neighbors))
-                 .data_ptr<int64_t>()
+          ? [&] {
+            int64_t nonzero_count = 0;
+            AT_DISPATCH_ALL_TYPES(
+                probs_or_mask.value().scalar_type(), "NumPick", ([&] {
+                  scalar_t* probs_data_ptr = probs_or_mask.value().data_ptr<scalar_t>() + offset;
+                  for (auto i = offset; i < offset + num_neighbors; ++ i, ++ probs_data_ptr) {
```

Comment:
We can use std::count here, https://en.cppreference.com/w/cpp/algorithm/count
we can count how many 0s there are by passing 0 to value parameter of std::count and subtract from num_neighbors.